let main =
  if (failure 0) {
    print error_message
  }